src/utils/formatNumbers.ts

```diff
@@ -4,16 +4,32 @@ import { DEFAULT_LOCALE } from 'constants/locales'
 import numbro from 'numbro'
 
 // Convert [CurrencyAmount] to number with necessary precision for price formatting.
-export const currencyAmountToPreciseFloat = (currencyAmount: CurrencyAmount<Currency>) => {
-  const floatForLargerNumbers = parseFloat(currencyAmount.toFixed(3))
+export const currencyAmountToPreciseFloat = (currencyAmount: CurrencyAmount<Currency> | undefined) => {
+  if (!currencyAmount) return undefined
+  const floatForLargerNumbers = parseFloat(currencyAmount.toExact())
   if (floatForLargerNumbers < 0.1) {
-    return parseFloat(currencyAmount.toSignificant(3))
+    return parseFloat(currencyAmount.toSignificant(6))
   }
-  return floatForLargerNumbers
+  return parseFloat(currencyAmount.toExact())
```

Comment:
aren't you doing this above? could you just return `floatForLargerNumbers`?

Reply:
yes omg sorry, rushed botched copy pasta.